TinCanTests/ActivityTest.cs

```diff
@@ -40,12 +40,16 @@ public void TestActivityIdCase()
         }
 
         [Test]
-        [ExpectedException("System.UriFormatException")]
         public void TestActivityIdInvalidUri()
         {
-            var activity = new Activity();
-            string invalid = "foo";
-            activity.id = invalid;
+            Assert.Throws<System.UriFormatException>(
```

Comment:
In cases where there is an assertion function specific to what the test is doing I was thinking the more specific variant ought to be used. `Assert.That` seems like a generic wrapper which makes sense for other cases. `Assert.Throws` Avoids `InstanceOf`, etc. I didn't find anything definitive but given 3.x had other breaking changes, they just added `Assert.ThrowsAsync`, and it is decently documented I think the more specific one is fine (or better).